src/Mvc/Mvc.ViewFeatures/src/RazorComponents/MvcRazorComponentPrerenderer.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Text.Encodings.Web;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Components.Rendering;
+using Microsoft.AspNetCore.Components.Server;
+using Microsoft.AspNetCore.Components.Services;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures.RazorComponents
+{
+    internal class MvcRazorComponentPrerenderer : IComponentPrerenderer
+    {
+        private HtmlEncoder _encoder;
+
+        public MvcRazorComponentPrerenderer(HtmlEncoder encoder)
+        {
+            _encoder = encoder;
+        }
+
+        public async Task<IEnumerable<string>> PrerenderComponentAsync(ComponentPrerenderingContext context)
+        {
+            var dispatcher = Renderer.CreateDefaultDispatcher();
+            var parameters = context.Parameters;
+            var helper = (HttpUriHelper)context.Context.RequestServices.GetRequiredService<IUriHelper>();
```

Comment:
Can you add a comment that says this is intentional because we want a request scoped service and this shouldn't be moved to the ctor?

Reply:
Done